### PR TITLE
JS SDK Create an Integration

### PR DESCRIPTION
**Pull request by crystalqdm**
# Ticket: [WEB-3](https://raygun.atlassian.net/browse/WEB-3?atlOrigin=eyJpIjoiNmUzOTAwNmRjNTdjNGIwNTgzZDM0YWQ4NzZkYjA2ZGQiLCJwIjoiaiJ9)

## Description:

In conjunction with these PRs: [Public API](https://github.com/MindscapeHQ/raygun/pull/14989) | [.NET SDK](https://github.com/MindscapeHQ/Raygun.API.SDK/pull/50)

This PR implements create-integration functionality within the Javascript SDK.
(NOTE: changes from [this ](https://github.com/MindscapeHQ/raygunapi-js/pull/11)ticket may appear in this branch too until its branch has been merged to main)

## Screenshots:
Response from Create Integration in JS SDK
image

## Test plan:
- [x] Test endpoints via local console app
- [x] Test endpoints via js sdk test browser
- [x] Test via Postman

### Author to check:
- [ ] Builds pass 
- [ ] Tested in an alternative environment
- [ ] Reviewed by another developer
- [ ] Appropriate documentation written

### Reviewer to check:
- [ ] Change has been tested on Beta
- [ ] Code is written to standards
- [ ] Appropriate tests have been written

[Engineering docs](https://raygunners.quip.com/ah5PA7OHZaUk)

### CODE REVIEW

File: lib/entities/integrations/index.ts [hunk 1]
```diff
@@ -19,11 +20,30 @@ export class Integrations {
...
 
-       return await wrapWithErrorHandler(async () => {
-          const url = buildApiUrl(urlSegments);
-          const res = await this.networkClient.get<IIntegration[]>(url);
-          return res;
-      });
+  /**
+   * Create a new integration for the given plan
+   * @param planIdentifier - Identifier of the target plan
+   * @param integration - The integration to create
```
Comment (sumitramanga) [inline lib/entities/integrations/index.ts]: I think the integration param could be more descriptive such as "The integration payload to create a new integration"

File: lib/entities/integrations/index.ts [hunk 2]
```diff
@@ -19,11 +20,30 @@ export class Integrations {
...
+   * @returns The created integration
+   */
+  async create(planIdentifier: string, integration: ICreateIntegrationPayload): Promise<IIntegration | undefined> {
```
Comment (sumitramanga) [inline lib/entities/integrations/index.ts:38]: Was the type undefined for the Promise for creating an alert?

**Review by sumitramanga — CHANGES_REQUESTED** [review]
Nice one!! Just a couple of things from me

File: lib/entities/integrations/index.ts (hunk 2 above)
Comment (crystalqdm) [reply lib/entities/integrations/index.ts:38]: yep! so does applications, so I figured I would keep this consistent in integrations too

**Review by sumitramanga — APPROVED** [review]
Nice, thank you for the changes! Just one more thing I forgot to point out. You will need to bump the version of this within the package.json. This is for NPM purposes (when we publish this to NPM, it will want a new version number). You just want to increment the last number

Otherwise, everything looks good! Just hold off merging after we've deployed everything successfully :)

**Pull request merged**